app/models/manageiq/providers/openstack/infra_manager/host.rb

```diff
@@ -203,4 +203,112 @@ def disconnect_ems(e = nil)
     self.availability_zone = nil if e.nil? || ext_management_system == e
     super
   end
+
+  def supports_start?
+    (state.casecmp("off") == 0 && hardware.provision_state == "active")
+  end
+
+  def validate_start
+    if supports_start?
+      {:available => true,   :message => nil}
+    else
+      {:available => false,  :message => _("Cannot start. Already on.")}
+    end
+  end
+
+  def start(userid = "system")
+    ironic_set_power_state_queue(userid, "power on", "Starting", "start", :host_start)
+  end
+
+  def supports_stop?
+    state.casecmp("on") == 0
+  end
+
+  def validate_stop
+    if supports_stop?
+      {:available => true,   :message => nil}
+    else
+      {:available => false,  :message => _("Cannot stop. Already off.")}
+    end
+  end
+
+  def stop(userid = "system")
+    ironic_set_power_state_queue(userid, "power off", "Stopping", "stop", :host_stop)
+  end
+
+  def ironic_set_power_state_queue(userid = "system",
+                                   power_state = "power offf",
+                                   power_state_text_verb = "Stopping",
+                                   power_state_text = "stop",
+                                   feature = :host_stop)
+    log_target = "#{self.class.name} name: [#{name}], id: [#{id}]"
+
+    task = MiqTask.create(:name => "#{power_state_text_verb} Ironic node '#{name}'", :userid => userid)
+
+    $log.info("#{power_state_text_verb} Ironic node #{log_target}")
```

Comment:
You want `_log` instead of `$log` here (and other places throughout) as it will do appropriate formatting of the log message